tests/src/test/scala/tests/TypoSuite.scala

```diff
@@ -194,21 +194,27 @@ class TypoSuite extends FunSuite {
 
   // extension methods to abstract over lsp4s service creation.
   implicit class XtensionServices(services: Services) {
-    def forwardRequest[A, B](endpoint: Endpoint[A, B])(implicit client: JsonRpcClient): Services = {
-      services.requestAsync(endpoint)(a => endpoint.request(a))
+    def forwardRequest[A, B](endpoint: Endpoint[A, B])(implicit client: RpcClient): Services = {
+      //services.requestAsync(endpoint)(a => endpoint.request(a))
+
+      // FIXME  returns a Task[RpcReponse[B]] but needs a Task[B]
+      // Not fully sure how to fix this one
+
+      //services.requestAsync(endpoint)(a => endpoint.request(a))
+      ???
```

Comment:
```suggestion
      services.requestAsync(endpoint) { a =>
        endpoint.request(a).flatMap {
          case RpcResponse.RpcSuccess(value, _) => Task.now(value)
          case err: RpcResponse.RpcFailure      => Task.raiseError(err)
        }
      }
```

Reply:
🤔 so I'm still struggling with this a bit. While this compiles find I can't figure out why this keeps timing out.

https://github.com/build-server-protocol/build-server-protocol/runs/2700373516?check_suite_focus=true#step:7:81

Any ideas?

Reply:
I'm trying to look at this tomorrow, sorry for the delay @ckipp01

Reply:
I'm so sorry Chris, as much as I try I can't get to take a look at this. Have you looked at the implementation of the tests and maybe even [jsonrpc4s](https://github.com/jvican/jsonrpc4s)? I don't think this should time out so maybe some exception is not being thrown properly or the tests wait on a promise that is never completed?

Reply:
Hey, no worries at all. When I get a bit of free time I'll do a deeper dive into jsonrpc4s to see what's going on. From looking at the tests in here it wasn't really clear why this would happen, but I could totally be missing something that has changed between the lsp4s and jsonrpc4s implementation. I'll dig deeper.